src/core/sock.c

```diff
@@ -313,6 +314,11 @@ static int nn_sock_setopt_inner (struct nn_sock *self, int level,
                 return -EINVAL;
             dst = &self->rcvbuf;
             break;
+        case NN_RCVMAXSIZE:
+            if (nn_slow (val <= -1))
```

Comment:
This was a typo, good catch. Should be `(val < -1)`. I think "negative but not -1" should be reserved and thus an EINVAL. I'll fix and also add a test.